subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -190,36 +186,21 @@ WatchingAwareVirtualFileSystem createVirtualFileSystem(
                 updateFunctionDecorator,
                 DirectoryScanner.getDefaultExcludes()
             );
-            WatchingAwareVirtualFileSystem watchingAwareVirtualFileSystem = determineWatcherRegistryFactory(OperatingSystem.current())
-                .<WatchingAwareVirtualFileSystem>map(watcherRegistryFactory -> new WatchingVirtualFileSystem(
-                    watcherRegistryFactory,
-                    delegate,
-                    updateFunctionDecorator,
-                    watchFilter,
-                    sectionId -> documentationRegistry.getDocumentationFor("gradle_daemon", sectionId)
-                ))
-                .orElse(new NonWatchingVirtualFileSystem(delegate));
+            WatchingAwareVirtualFileSystem watchingAwareVirtualFileSystem = fileWatcherRegistryFactory instanceof UnavailableFileWatcherRegistryFactory
+                ? new NonWatchingVirtualFileSystem(delegate)
+                : new WatchingVirtualFileSystem(
+                fileWatcherRegistryFactory,
+                delegate,
+                updateFunctionDecorator,
+                watchFilter,
+                sectionId -> documentationRegistry.getDocumentationFor("gradle_daemon", sectionId)
+            );
             listenerManager.addListener((BuildAddedListener) buildState ->
                 watchingAwareVirtualFileSystem.buildRootDirectoryAdded(buildState.getBuildRootDir())
             );
             return watchingAwareVirtualFileSystem;
         }
```

Comment:
I see that we are now returning a `WatchingAwareVirtualFileSystem` here, but is that really necessary? Can we somehow not expose more than `VirtualFileSystem`?